atd-vzv/src/views/summary/helpers/helpers.js

```diff
@@ -2,42 +2,71 @@ import { lifespanYears } from "../../../constants/calc";
 import { dataEndDate } from "../../../constants/time";
 // Helpers to handle Socrata responses for Summary view components
 
-export const calculateTotalFatalities = data =>
+const calcFieldTotalsFromRecords = (data, prevYear, currentYear, field) =>
   data.reduce(
-    (accumulator, record) => (accumulator += parseInt(record.death_cnt)),
-    0
+    (accumulator, record) => {
+      const recordYear = record.crash_date.slice(0, 4);
+      accumulator = {
+        ...accumulator,
+        [recordYear]: (accumulator[recordYear] += parseInt(record[field])),
+      };
+      return accumulator;
+    },
+    { [prevYear]: 0, [currentYear]: 0 }
```

Comment:
Very clever stuff, I had been wondering how you separate the data for two years by year, figured you would have some kind of key-value object somewhere 😹 

Thank you for showing me creative ways of using map/reduce in javascript. 👍

Reply:
🧠

Reply:
Haha, yes, I was happy to get these functions a little meta. I agree about further optimization in your comment below, @sergio. It felt a little unfinished after refactoring each individual function.